### ISSUE TITLE
fix current/previous/next playlist building when current clip is blank media

### ISSUE THREAD

**Issue by Lacrymology** [inaes-tic/mbc-mosto#107]
Current playlist should show `"INVALID"` and previous and next playlists should be found by date

**Comment by Lacrymology** [comment]
with the current setup, this should be handled by caspa.

When mosto doesn't know about the show (it's not loaded in the playlists collection), it sends an empty  object `{}`, when it's an invalid show (a clip with no playlist, like BlankClip), it sends `_id: -1`, otherwise, it sends `_id: <schedule id>`

I'm not convinced this is the right approach, but a communications protocol was never properly designed.

In any case, this will have to wait until the new caspa is merged

**Comment by jmrunge** [comment]
Why does caspa never (or almost never) show previous clip and previous
show?  Are we sure we are sending them? (thats why I left this issue
opened, I wanted to make sure of this).

2013/9/17 Tomas Neme notifications@github.com

**Comment by Lacrymology** [comment]
no, we're not, because in most cases previous clip / show are not loaded into mosto. The fact that invalid show sends `id = -1` and unknown sends `{}` is so caspa can look for the expected previous show.

I don't know what's the right behavior here. If we should show the historically previous show, whatever the programming says, we need to keep track of this somewhere, and keep it between mosto reboots, and whatnot. Right now what's being sent is what mosto has loaded in it's Playlists and MeltedMedia backbone collections.
